fixtures/enum-types/src/lib.rs

```diff
@@ -0,0 +1,41 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+enum Animal {
+    Dog,
+    Cat,
+}
+
+// Though it has the proc-macro, we drop the variant
+// literals if there is not a repr type defined
+#[derive(uniffi::Enum)]
+pub enum AnimalNoReprInt {
+    Dog = 3,
+    Cat = 4,
+}
+
+#[repr(u8)]
+#[derive(uniffi::Enum)]
+pub enum AnimalUInt {
+    Dog = 3,
+    Cat = 4,
+}
+
+#[repr(u64)]
+#[derive(uniffi::Enum)]
+pub enum AnimalLargeUInt {
+    Dog = 123456789,
+    Cat = 987654321,
+}
+
+// Signed is currently NOT supported but included to ensure we don't break
+// things for current users of the enum proc-macro
+#[repr(i8)]
+#[derive(uniffi::Enum)]
+pub enum AnimalSignedInt {
```

Comment:
I think all tests should reference this. I think for Python it will work correctly. For kotlin and swift it should represent how it *does not* work - ie, somehow that `.value` doesn't exist for this enum.